### PR TITLE
Docs: Add a table of contents to the top of the docs

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@nalimilan Looks like CI is not running on this PR?

**Comment by nalimilan** [comment]
Thanks. CI no longer works since it was on travis-ci.com. We should move to GitHub Actions.

File: docs/src/index.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 The Statistics standard library module contains basic statistics functionality.
 
+## Index
+
+```@index
+Modules = [Statistics]
+```
+
+## Docs
```
Comment (nalimilan) [inline docs/src/index.md:11]: "Docs" might be a bit familiar? Anyway I think we should standardize on a common organization and wording across all of Julia. Probably better discuss this in a large PR in the Julia repo first.

**Comment by DilumAluthge** [comment]
Closing in favor of first discussing this in the Julia repo.

**Pull request closed without merging**